proto/engine/v1/json_marshal_unmarshal.go

```diff
@@ -109,6 +147,82 @@ func (b *PayloadIDBytes) UnmarshalJSON(enc []byte) error {
 	return nil
 }
 
+type withdrawalJSON struct {
+	Index     *hexutil.Uint64 `json:"index"`
+	Validator *hexutil.Uint64 `json:"validatorIndex"`
+	Address   *common.Address `json:"address"`
+	Amount    string          `json:"amount"`
+}
+
+func (j *withdrawalJSON) ToWithdrawal() (*Withdrawal, error) {
+	w := &Withdrawal{}
+	b, err := json.Marshal(j)
+	if err != nil {
+		return nil, err
+	}
+	if err := w.UnmarshalJSON(b); err != nil {
+		return nil, err
+	}
+	return w, nil
+}
+
+func (w *Withdrawal) toWithdrawalJSON() (*withdrawalJSON, error) {
```

Comment:
This is an odd method, why do we have it ?

Reply:
I use it here:
```
ws := make([]*withdrawalJSON, len(e.Withdrawals))
for i, w := range e.Withdrawals {
	ws[i], err = w.toWithdrawalJSON()
	if err != nil {
		return nil, err
	}
}
decoded["withdrawals"] = ws
```